### DISCUSSION TITLE
How to get response of the payment after showing presentPaymentSheet ?

### DISCUSSION THREAD

**Discussion by h1amza** [flutter-stripe/flutter_stripe#1167] [Q&A]
This is my code :

First Step:
```
await Stripe.instance.initPaymentSheet(
            paymentSheetParameters: SetupPaymentSheetParameters(
              customFlow: true,
              merchantDisplayName: 'Flutter Stripe Store Demo',
              paymentIntentClientSecret: paymentIntentData!['clientSecret'],
            ),
          );
```

Second:
```
{
    try {
      debugPrint("display");
      await Stripe.instance.presentPaymentSheet();
      debugPrint("end display");

    } on Exception catch (e) {
      if (e is StripeException) {
        debugPrint("Error from Stripe: ${e.error.localizedMessage}");
      } else {
        debugPrint("displayPaymentSheet Unforeseen error: $e");
      }
    } catch (e) {
      debugPrint("displayPaymentSheet exception:$e");
    }
  }
```

So how to get paymentSheetResponse to Handle the result and get id off the payment and status?

**Comment by h1amza** [comment]
@remonh87 @jonasbark can you help plz or show me in example for that, i already see this: https://www.youtube.com/watch?v=pGkpOZXrcU4 , and read https://docs.page/flutter-stripe/flutter_stripe/sheet

**Comment by remonh87** [comment]
The payment sheet completes in case the data is collected correctly. If some error happens the library Throws a stripe exception. There is not yet proof the payment is successful so you need to ask your backend if they received success from Stripe. Also check: https://docs.page/flutter-stripe/flutter_stripe/sheet#4--handle-post-payment-events

**Comment by h1amza** [comment]
Ok, I will see, thanks for your quick responding
